torch/fx/experimental/Partitioner.py

```diff
@@ -180,6 +335,101 @@ def create_partition(self) -> Partition:
         """Create a partition and append it to self.partitions."""
         partition_id = len(self.partitions)
         assert isinstance(self.graph_module, GraphModule)
-        partition = Partition(partition_id, self.graph_module)
+        partition = Partition(partition_id, self.graph_module, self.orphans)
         self.partitions.append(partition)
         return partition
+
+    def combine_partitions_based_on_size(self, available_mem) -> None:
+        """Combining small partitions together to keep as less partitions as possible.
+           Here is an example of the algorithm to do this:
+           Assume some partitions, we first sort them based on partiiton used memory size.
+           [(partition_4, 1), (partition_3, 1), (partition_2, 2), (partition_1, 7), (partition_0, 9)]
+           The available memory is 10.
+
+           step 1: self.find_partition_to_combine_based_on_size()
+           First, look the smallest partition, partition_4: 10 - 1 = 9
+           It means any partition has a used memory equal or less than 9 could combine this partition
+           We go from the largest and selection partition_0.
+           As a result, partition_9 has a size of 10
+
+           Then, repeat step 1, for partition_3, 10 - 1 = 9
+           partition_1 has a size of 7 and could combine with partition_3
+           For partition_2, 10 - 2 = 8
+           At this moment, partition_1 has a size of 8 and it combines partition_2
+        """
```

Comment:
High-level question: doesn't combining partitions need to do some sort of analysis of the data dependencies between partitions? For example, take the following graph:

```
    A
   / \
  B   C
  |  / \
  | /  |
  D    E
```

Partitions are assigned as:
Partition 0 : {A}
Partition 1: {B, C}
Partition 2: {D, E}

Suppose this algorithm determines that partition 0 can be combined with partition 2, so:

Partition 0 : {A, D, E}
Partition 1: {B, C}

However, this would create a circular dependency between partitions

Partition0(B, C) -> (A, D, E):
  A = op()
  D = op(B, C)
  E = op(C)
  return A, D, E

Partition1(A) -> (B, C):
  B = op(A)
  C = op(A)
  return B, C

Reply:
Yes, it should. Will add it.